lib/filterx/expr-comparison.c

```diff
@@ -149,11 +148,11 @@ _evaluate_type_aware(FilterXObject *lhs, FilterXObject *rhs, gint operator)
   if (lhs->type == rhs->type &&
       (filterx_object_is_type(lhs, &FILTERX_TYPE_NAME(string)) ||
        filterx_object_is_type(lhs, &FILTERX_TYPE_NAME(bytes)) ||
-       filterx_object_is_type(lhs, &FILTERX_TYPE_NAME(protobuf)) ||
-       filterx_object_is_type(lhs, &FILTERX_TYPE_NAME(json_object)) || // TODO: we should have generic map and array cmp
-       filterx_object_is_type(lhs, &FILTERX_TYPE_NAME(json_array))))
+       filterx_object_is_type(lhs, &FILTERX_TYPE_NAME(protobuf))))
     return _evaluate_as_string(lhs, rhs, operator);
 
+  /* FIXME: add dict/list comparison */
```

Comment:
Can we merge it without comparison? We probably don't have too many jsons/dicts compared at the moment, maybe just equals/notequals.

Reply:
Yes, after the release.

This way, we have time to implement the comparison separately.